rust/src/shard.rs

```diff
@@ -39,8 +39,12 @@ use std::ffi::CString;
 use std::os::raw::c_char;
 
 pub trait Shard {
-  fn registerName() -> &'static str;
-  fn hash() -> u32;
+  fn registerName() -> &'static str
+  where
+    Self: Sized;
```

Comment:
Wouldn't it be easier to derive `Shard` from `Sized` instead of marking every single function?

```rust
pub trait Shard: Sized {
  // ...
}
```

Reply:
Wouldn't work apparently cos those calls are static  :)